packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/pre-push-cfn-modifier.ts

```diff
@@ -0,0 +1,29 @@
+import Resource from 'cloudform-types/types/resource';
+import _ from 'lodash';
+import { applyS3SSEModification } from './modifiers/s3-sse-modifier';
+import { Template } from 'cloudform-types';
+
+// modifies the template in-place
+export type TemplateModifier = (template: Template) => Promise<void>;
+
+// modifies the resource in-place
+export type ResourceModifier = (resource: Resource) => Promise<void>;
+
+export const prePushCfnTemplateModifier: TemplateModifier = async template => {
+  await Promise.all(
+    Object.values(template.Resources).map(async resource => {
+      const transformers = getResourceModifiers(resource.Type);
+      await Promise.all(transformers.map(transformer => transformer(resource)));
```

Comment:
How are you ensuring that there is no racecondition and a modified resource is not overridden by another one?

Reply:
I can run them sequentially but that's still not going to prevent one transformer from stepping on the toes of the previous one. Each transformer should be designed in a way that it doesn't write over the changes of another

Reply:
Per discussion offline, serializing execution of modifiers and passing in the previous resource into the next